torch/csrc/autograd/python_variable.cpp

```diff
@@ -154,6 +154,13 @@ static bool check_has_torch_dispatch(PyObject *obj) {
   );
 }
 
+static PyObject* getXLATensorClass() {
+  // TODO: What should we do if XLATensor doesn't exist? Return THPVariableClass?
+  // TODO: is THPObjectPtr the right pointer container?
+  static THPObjectPtr xla_tensor_ptr(py::globals()["XLATensor"].ptr());
+  return xla_tensor_ptr.get();
```

Comment:
It's better to leak the PyObject as the destructor will run at shutdown when the Python interpreter may have already exited. See also how THPVariableClass is implemented.

Note that unlike THPVariableClass we cannot assume that the XLATensor is available at PyTorch time, since it doesn't show up until XLA loads itself. So it might be more appropriate to have some private C api for "Registering" the class, and then having XLA register the class itself at load time

Reply:
> Note that unlike THPVariableClass we cannot assume that the XLATensor is available at PyTorch time, since it doesn't show up until XLA loads itself. 

I was thinking that the XLA extension is loaded before we could start creating XLA tensors, so hopefully, we can work with XLA guys to make sure `XLATensor` appears in `globals()` before we would try to create XLA tensors?

Also, I wonder if it would be acceptable to fallback to `THPVariableClass` if `XLATensor` isn't available? 

> So it might be more appropriate to have some private C api for "Registering" the class, and then having XLA register the class itself at load time

hehe, this sounds like a proper solution you mentioned @albanD might be working on :) I could work with him if we can't figure out an acceptable hack.

Reply:
> I was thinking that the XLA extension is loaded before we could start creating XLA tensors, so hopefully, we can work with XLA guys to make sure XLATensor appears in globals() before we would try to create XLA tensors?

I guess these are both valid ways to skin the cat. But it feels weird for PyTorch to reach out to a module that doesn't exist anywhere in PyTorch: you call me versus I'll call you.

Reply:
fair enough! I put together another quick sketch. I wonder what you and @albanD  think